pkg/providers/tinkerbell/reconciler/reconciler.go

```diff
@@ -256,3 +259,44 @@ func (r *Reconciler) ValidateHardware(ctx context.Context, log logr.Logger, clus
 
 	return controller.Result{}, nil
 }
+
+// ValidateRufioMachines checks to ensure all the Rufio machines condition contactable is True.
+func (r *Reconciler) ValidateRufioMachines(ctx context.Context, log logr.Logger, clusterSpec *c.Spec) (controller.Result, error) {
+	log = log.WithValues("phase", "validateRufioMachines")
+
+	kubeReader := hardware.NewKubeReader(r.client)
+	if err := kubeReader.LoadRufioMachines(ctx); err != nil {
+		log.Error(err, "rufio machine check failure")
+		failureMessage := err.Error()
+		clusterSpec.Cluster.Status.FailureMessage = &failureMessage
+
+		return controller.ResultWithReturn(), nil
+	}
+
+	for _, rm := range kubeReader.GetCatalogue().AllBMCs() {
+		if err := r.checkContactable(rm); err != nil {
```

Comment:
So is this validating that all the rufio machines are contactable before reconciling the cluster? Or only the new rufio machines that are added?

Reply:
Actually it checks all. Thats what i wanted to discuss here.

https://github.com/aws/eks-anywhere/pull/4908#discussion_r1104126052